forc-plugins/forc-doc/src/doc.rs

```diff
@@ -160,26 +161,35 @@ impl Document {
     }
 }
 impl Renderable for Document {
-    fn render(self) -> Box<dyn RenderBox> {
-        box_html! {
-            : self.item_header.render();
-            : self.item_body.render();
-        }
+    fn render(self) -> Result<Box<dyn RenderBox>> {
+        let header = self.item_header.render()?;
+        let body = self.item_body.render()?;
+        Ok(box_html! {
+            : header;
+            : body;
+        })
     }
 }
+
 pub(crate) type ModulePrefix = String;
 #[derive(Clone, Ord, PartialOrd, Eq, PartialEq)]
 pub(crate) struct ModuleInfo(pub(crate) Vec<ModulePrefix>);
 impl ModuleInfo {
     /// The current module.
+    ///
+    /// Panics if there are no modules.
     pub(crate) fn location(&self) -> &str {
         self.0
             .last()
-            .expect("There will always be at least the project name")
+            .expect("Expected Some module location, found None")
```

Comment:
Sorry if I wasn't clear with what I meant by "going back to our original matching", but this could be what you originally went with in your initial PR:

```rust
match self.0.last() {
  Some(location) => Ok(location),
  None => bail!("There will always be at least the module location"),
```

Note also the err message change, as pointed out [here](https://github.com/FuelLabs/sway/pull/3691#discussion_r1073313586).

Reply:
Hmm I think for now the expect message here is okay. Implementing result there means you'd have to worry about how you handle every call to location in the render macro which doesn't return a result. There's a few for loops in there where I can see it would be difficult to find a non-hacky way of resolving the results before using them in the `box_html!` macro.

Reply:
that's reasonable!